starboard/build/toolchain/BUILD.gn

```diff
@@ -13,8 +13,59 @@
 # limitations under the License.
 
 import("//build/config/clang/clang.gni")
+import("//build/config/win/visual_studio_version.gni")
 import("//build/toolchain/gcc_toolchain.gni")
+import("//starboard/build/config/os_definitions.gni")
 
-clang_toolchain("clang") {
-  clang_base_path = clang_base_path
+# Use this toolchain for linux-x64 platforms.
+if (is_linux && !using_old_compiler && target_cpu == "x64") {
+  clang_toolchain("clang") {
+    clang_base_path = clang_base_path
+  }
+} else {
+  import("//$starboard_path/toolchain/variables.gni")
+  assert(
+      defined(native_linker_path),
+      "native_linker_path should be defined in $starboard_path/toolchain/variables.gni")
+}
+
+# Use this toolchain for raspi platforms, old linux compiler platforms.
+if ((is_linux && target_cpu == "arm") || using_old_compiler) {
+  gcc_toolchain("clang") {
+    prefix = rebase_path("$clang_base_path/bin", root_build_dir)
+    cc = "$prefix/clang"
+    cxx = "$prefix/clang++"
+    ld = native_linker_path
+    readelf = "readelf"
+    ar = "${prefix}/llvm-ar"
+    nm = "nm"
+    toolchain_args = {
+      is_clang = true
+    }
+  }
+}
+
+# Use this toolchain for Windows based platforms.
+if (is_win) {
+  gcc_toolchain("clang") {
```

Comment:
This block looks very similar to the one above except for `.exe` program suffixes - which i believe are a predefined GN variable ( e.g. `default_output_extension` )

readelf vs llvm-readobj are still different though.

We could _try_ and collapse those 2 down to one, but i don't actually know if this this will hurt or help readability

Reply:
readelf and nm are different , 

and it seemed more trouble to unify them like for example in the unified toolchain you would either add 

if(defined (native_executable_extension). &&
if(defined(shlib_extension)  or add those variables to raspi variables.gni

im not sure if there any other windows specific variables - like for xbox, win32 : we might decide on the windows clang compatible toolchain as that's what linux uses - that would add another block for cc, cxx 

i thought for now it's best to keep the windows one separate.

Also i can address this and other comments from yavor, andrew in a follow up cl - so dunno if this should be blocking

Reply:
Yeah it's fine as separate. Trying to consolidate too much would result in less readable outcome here.

Reply:
I think these should be consolidated—picking the tool name can either be done with variables whose values are conditional on the current_os, then used in a single toolchain definition, or it can be done with an intermediate GN template, which is probably my preferred solution.